modvendor/modvendor.go

```diff
@@ -0,0 +1,223 @@
+// Copyright 2022 Mineiros GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package modvendor
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/mineiros-io/terramate/errors"
+	"github.com/mineiros-io/terramate/git"
+	"github.com/rs/zerolog/log"
+)
+
+// Source represents a module source
+type Source struct {
```

Comment:
we have this already here: https://github.com/mineiros-io/terramate/blob/main/tf/terraform.go#L31
I think this code should be under a "terraform related package".

Reply:
Moved to tf